### DISCUSSION TITLE
Is it possible to overlap control drum mesh with reactor core lattice?

### DISCUSSION THREAD

**Discussion by robertjdemuth** [idaholab/moose#25539] [Q&A Meshing]
I am new to MOOSE and using the reactor module to mesh a control drum and hexagonal lattice reactor core. The reactor I am trying to model has already been developed and optimized by a previous student in SCALE. However, I am running into an issue creating the mesh with the reactor module because in the SCALE model, the control drum and hexagonal lattice overlap as seen below:
![image](https://github.com/idaholab/moose/assets/145687537/03b58c79-0fde-44e4-b73a-a62398bc5e1e)
I can't find anything in the docs where you can delete sections of the hexagonal lattice and replace it with another mesh. I have created the model using the PatternedHexMeshGenerator to create the lattice and PeripheralRingMeshGenerator to create the control drum and combined them using XYDelaunayGenerator.
![image](https://github.com/idaholab/moose/assets/145687537/c8f4a49a-1678-45cd-a5e0-234fac4815d4)
Is this even possible using the reactor module, or would I need to import a geometry meshed using a different software?

**Comment by GiudGiud** [comment]
@miaoyinb

**Comment by miaoyinb** [comment]
If the background region (the region outside the central circles) of the partial assembly meshes are the same material (block) as the peripheral region. You can use [`FlexiblePatternGenerator`](https://mooseframework.inl.gov/source/meshgenerators/FlexiblePatternGenerator.html) to do it.

If not, meshing it might still be possible but the process would be complex/non-trivial. We can discuss if needed.

**Reply by robertjdemuth** [reply]
The background region of the partial assembly meshes is the same block as the peripheral region, however, there is a duct on the hexagonal lattice with a different block. Would this be an issue?

**Reply by miaoyinb** [reply]
Is the duct connected to the control drum?

**Reply by robertjdemuth** [reply]
The duct is around the hexagonal lattice structure for the fuel pin. The control drum does not have a duct.

**Reply by miaoyinb** [reply]
As your control drum is overlapped with the hexagon structure. At the position of the overlapping, is the duct of the hexagon structure connected to the external surface of the control drum?

**Reply by robertjdemuth** [reply]
The current model in scale just has the duct disappear at the location of intersection, but it could be added or removed. Our design is not at that level of detail yet.

**Reply by miaoyinb** [reply]
Whether the duct needs to be connected to the control drum or not would actually affect the complexity in meshing.

If they are not connected, you may use `XYMeshLineCutter` to cut a fraction of the hexagon structure and then use `FlexiblePatternGenerator` to fill the gap. Otherwise, I think a non-trivial use of `ParsedCurveGenerator` + `XYDelaunayGenerator` may be needed to take care of things.
